modules/history/src/views/full/MessageViewer.jsx

```diff
@@ -76,84 +24,157 @@ function NoConversationSelected() {
+            </div>
+            <div>
+              <IoMdFlag
+                className={style.messageTaskBarUnflagIcon}
+                data-tip
+                data-for="unflag"
+                onClick={() => this.props.unflag()}
+              />
+              <ReactTooltip id="unflag" effect="solid">
+                <div>Unflag Selected messages</div>
+              </ReactTooltip>
             </div>
           </div>
         )}
-        <div className={style['message-header-icons']}>
-          <div className={style['message-header-icon_item']}>
-            <a
-              href={this.fileURL}
-              download="message_history"
-              style={{
-                color: '#233abc'
-              }}
-            >
-              <MdFileDownload />
-            </a>
-          </div>
-          <div className={style['message-header-icon_item']}>
-            <CopyToClipboard text={window.location.href}>
-              <FiLink data-tip data-event="mousedown" data-event-off="mouseup" />
-            </CopyToClipboard>
+        {this.props.useAsFilter && (
+          <div>
+            <span>Display only flagged messages:</span>
+            <input type="checkbox" checked={this.state.filters.flag} onChange={() => this.toggleFlagFilter()} />
           </div>
-        </div>
-        <ReactTooltip delayHide={500} effect="solid">
-          <div>copied</div>
-        </ReactTooltip>
+        )}
       </div>
     )
   }
 }
 
-function MessageInspector(props) {
-  return (
-    <div
-      className={classnames(style['message-inspector'], {
-        [style['message-inspector-hidden']]: props.inspectorIsShown
-      })}
-    >
-      <div className={style['quit-inspector']} onClick={props.closeInspector}>
-        <GoX />
-      </div>
-      {props.currentlyFocusedMessage && (
-        <JSONTree theme={inspectorTheme} data={props.currentlyFocusedMessage} invertTheme={false} hideRoot={true} />
-      )}
-    </div>
-  )
-}
-
 export class MessageViewer extends React.Component {
   state = {
     inspectorIsShown: false,
-    currentlyFocusedMessage: null
+    currentlyFocusedMessage: null,
+    areMessagesSelected: false,
+    areAllMessagesSelected: false,
+    filters: { flag: false },
+    selectedGroups: [],
+    currentConversation: null
+  }
+
+  componentDidUpdate() {
+    if (this.props.conversation !== this.state.currentConversation) {
+      this.unselectAll()
+      this.setState({ currentConversation: this.props.conversation })
+    }
+  }
+
+  handleSelection(isSelected, group) {
+    let currentSelectedLength = this.state.selectedGroups.length
+    if (isSelected) {
+      currentSelectedLength += 1
+      this.selectMessage(group)
+    } else {
+      currentSelectedLength -= 1
+      this.unselectMessage(group)
+    }
+    this.setState({
+      areMessagesSelected: currentSelectedLength >= 0,
+      areAllMessagesSelected: currentSelectedLength >= this.props.messageGroups.length
+    })
```

Comment:
That is a computed property and not a state. If you want to use those properties without calling a method, try get accessors: 

```js
get areMessageSelected(){
 return this.state.selectedGroups.length >= 0
}

// and use like this:
if(this.areMessageSelected){}
```